### PR TITLE
Layout map item overviews position can be tweaked

### LINKED ISSUE #3292: [FEATURE][layouts] Allow control over stacking position of map overview extents
Original commit: https://github.com/qgis/QGIS/commit/38450f0befdf3cb4fb93bbd7b6bc7b32cb759204 by nyalldawson

Allows overview extents to be drawn below map labels (the new default),
but also adds choices for drawing them below all map layers, or
above/below a specific map layer.

This allows users to control exactly where in the map item's layer stack
the overview will be placed, e.g. allowing them to draw a overview
extent below some feature layers such as roads whilst drawing it
above other background layers.

### PR DESCRIPTION
**Pull request by DelazJ**
fix #3292

### CODE REVIEW

File: source/docs/user_manual/print_composer/composer_items/composer_map.rst [hunk 1]
```diff
@@ -351,22 +351,35 @@ You can:
...
+* The :guilabel:`Position` controls exactly where in the map item's layer stack
+  the overview will be placed, e.g. allowing them to draw a overview
+  extent below some feature layers such as roads whilst drawing it
+  above other background layers. Available options are:
+
+  * :guilabel:`Below map`
+  * :guilabel:`Below map layer` and :guilabel:`Above map layer`, selecting in
+    :guilabel:`Stacking layer` the layer to consider
+  * :guilabel:`Below map labels` and :guilabel:`Above map labels`: since
+    labels are drawn over all the features, the overview is placed respectively
```
Comment (havatv) [inline source/docs/user_manual/print_composer/composer_items/composer_map.rst]: Not sure I follow / understand this explanation, but then I am not too familiar with this.

Comment (DelazJ) [reply source/docs/user_manual/print_composer/composer_items/composer_map.rst]: (up to QGIS 3.10 but i'm not sure it has changed - i did not yet check the mask feature impact) when a map is rendered in QGIS, geometries of features are processed first. When all the layers have their features rendered, then their labels are rendered, so labels are always placed above geometry.
In this case, "below map labels" means below map labels but above map layers, and the second one means above everything in the map item. If it's clearer and you have a suggestion of rewording...

File: source/docs/user_manual/print_composer/composer_items/composer_map.rst [hunk 2]
```diff
@@ -351,22 +351,35 @@ You can:
...
-  item to center, when you have added several overviews.
+* The :guilabel:`Frame Style` uses the :ref:`symbol properties
+  <symbol-selector>` to render the overview frame.
+* The :guilabel:`Blending mode` allows you to set different transparency blend
+  modes.
+* The |checkbox| :guilabel:`Invert overview` creates a mask around the extents
+  when activated: the referenced map extents are shown clearly, whereas
+  everything else is blended with the frame fill color.
+* The |checkbox| :guilabel:`Center on overview` pans the map item content so
+  that the overview frame is displayed at the center of the map. You can only
```
Comment (havatv) [inline source/docs/user_manual/print_composer/composer_items/composer_map.rst]:
```suggestion
  that the overview frame is displayed at the center of the map.
  You can select this for as many overview maps as you like, but
```

File: source/docs/user_manual/print_composer/composer_items/composer_map.rst [hunk 3]
```diff
@@ -351,22 +351,35 @@ You can:
...
+  activate one overview item to center, when you have several overviews.
```
Comment (havatv) [inline source/docs/user_manual/print_composer/composer_items/composer_map.rst]:
```suggestion
  the centering will be according to the last one.
```

File: source/docs/user_manual/print_composer/composer_items/composer_map.rst [hunk 4]
```diff
@@ -351,22 +351,35 @@ You can:
...
+* The :guilabel:`Position` controls exactly where in the map item's layer stack
+  the overview will be placed, e.g. allowing them to draw a overview
```
Comment (havatv) [inline source/docs/user_manual/print_composer/composer_items/composer_map.rst]:
```suggestion
  the overview will be placed, e.g. allowing an overview extent to be
```

File: source/docs/user_manual/print_composer/composer_items/composer_map.rst [hunk 5]
```diff
@@ -351,22 +351,35 @@ You can:
...
+  extent below some feature layers such as roads whilst drawing it
```
Comment (havatv) [inline source/docs/user_manual/print_composer/composer_items/composer_map.rst]:
```suggestion
  drawn below some feature layers such as roads whilst drawing it
```

File: source/docs/user_manual/print_composer/composer_items/composer_map.rst (hunk 1 above)
Comment (havatv) [inline source/docs/user_manual/print_composer/composer_items/composer_map.rst]:
```suggestion
    labels are drawn above all the other map features, for these options
```

File: source/docs/user_manual/print_composer/composer_items/composer_map.rst [hunk 6]
```diff
@@ -351,22 +351,35 @@ You can:
...
+  above other background layers. Available options are:
+
+  * :guilabel:`Below map`
+  * :guilabel:`Below map layer` and :guilabel:`Above map layer`, selecting in
+    :guilabel:`Stacking layer` the layer to consider
+  * :guilabel:`Below map labels` and :guilabel:`Above map labels`: since
+    labels are drawn over all the features, the overview is placed respectively
+    above all the map layers or above the map item.
```
Comment (havatv) [inline source/docs/user_manual/print_composer/composer_items/composer_map.rst]:
```suggestion
    the overview is placed above all map contents except for
    the labels (if ``Below map labels`` chosen).
    For ``Above map labels``, the overview is placed above
    the labels too.
```

File: source/docs/user_manual/print_composer/composer_items/composer_map.rst (hunk 3 above)
Comment (DelazJ) [reply source/docs/user_manual/print_composer/composer_items/composer_map.rst]: Indeed, but this behavior is a bug IMO (https://github.com/qgis/QGIS/issues/33597): there's no interest in checking more than one (only the last wins and unchecking the last does not center on the penultimate so useless). And I'm not sure we should document buggy behavior

**Pull request merged**